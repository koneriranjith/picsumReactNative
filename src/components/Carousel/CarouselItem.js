import React from 'react';

import { View, StyleSheet, Text, Image, Dimensions} from 'react-native'

const { width, height} = Dimensions.get('window')


const CarouselItem = ({item}) => {
    const { id, author } = item || {};
    return (
        <View style={styles.cardView}>
            <View>
               <Text style={styles.author}>{author}</Text>
            </View>
            <Image style={styles.image} source={{ uri: `https://picsum.photos/200/300?image=${id}`}}/>
        </View>
    )
}

const styles = StyleSheet.create({
    cardView: {
        flex: 1,
        width: width - 20,
        height: height / 3,
        backgroundColor: 'white',
        margin: 10,
        shadowColor: '#000',
        shadowOffset: { width: 0.5, height: 0.5 },
        shadowOpacity: 0.5,
        shadowRadius: 3,
        elevation: 5
    },
    image: {
        width: width - 20,
        height: height /3
    },
    author: {
        fontSize: 22,
        // shadowColor
    }
})

export default CarouselItem;