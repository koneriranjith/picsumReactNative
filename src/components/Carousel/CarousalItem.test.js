import React from 'react';
import {shallow} from 'enzyme';
import CarouselItem from './CarouselItem';

describe('Carousel Item', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<CarouselItem item={{id:1, author:"author"}} />)
            expect(component).toMatchSnapshot()
            expect(component.find("Text")).toBeTruthy()
        });
    });
});
