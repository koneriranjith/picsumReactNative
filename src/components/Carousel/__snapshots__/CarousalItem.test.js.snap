// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Item Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
